src/client.ts

```diff
@@ -144,6 +142,20 @@ export default class Client {
     await this.start();
   }
 
+  async updateGem(): Promise<void> {
```

Comment:
One question I had while making this PR is how much of the functionality to include in the client vs in the `extension.ts` file. In the end, I chose to implement the state in the client and the display in `extension.ts`. Happy to change it and put all the functionality back in the client if that works better!